src/components/AddressHistory.js

```diff
@@ -0,0 +1,227 @@
+    if (!isEqual(this.state.queryParams, queryParams)) {
+      // Fetch new data, unless query params were cleared and we were already in the most recent page
+      if (queryParams.hash || this.state.hasBefore) {
+        this.getData(queryParams);
+      }
+    }
+  }
+
+  componentWillUnmount() {
+    WebSocketHandler.removeListener('network', this.handleWebsocket);
+  }
+
+  /**
+   * Called when 'network' ws message arrives
+   * If it's a new tx message update data, in case is necessary
+   *
+   * @param {Object} wsData Data from websocket
+   */
+  handleWebsocket = (wsData) => {
+    if (wsData.type === 'network:new_tx_accepted') {
+      this.updateListWs(wsData);
+    }
+  }
+
+  updateListWs = (tx) => {
+    // We only add new tx/blocks if it's the first page
+    if (!this.state.hasBefore) {
+      if (this.props.shouldUpdate(tx)) {
+        let transactions = this.state.transactions;
+        let hasAfter = (this.state.hasAfter || (transactions.length === TX_COUNT && !this.state.hasAfter))
+        transactions = helpers.updateListWs(transactions, tx, TX_COUNT);
+
+        let firstHash = transactions[0].tx_id;
+        let lastHash = transactions[transactions.length-1].tx_id;
+
+        // Finally we update the state again
+        this.setState({ transactions, hasAfter, firstHash, lastHash });
+      }
+    }
+  }
+
+  handleDataFetched = (data, queryParams) => {
+    // Handle differently if is the first GET response we receive
+    // page indicates if was clicked 'previous' or 'next'
+    // Set first and last hash of the transactions
+    let firstHash = null;
+    let lastHash = null;
+    if (data.transactions.length) {
+      firstHash = data.transactions[0].tx_id;
+      lastHash = data.transactions[data.transactions.length-1].tx_id;
+    }
+
+    let hasAfter;
+    let hasBefore;
+    if (queryParams.page === 'previous') {
+      hasAfter = true;
+      hasBefore = data.has_more;
+      if (!hasBefore) {
+        // Went back to most recent page: clear URL params
+        this.pagination.clearQueryParams();
+      }
+    } else if (queryParams.page === 'next') {
+      hasBefore = true;
+      hasAfter = data.has_more;
+    } else {
+      // First load without parameters
+      hasBefore = false;
+      hasAfter = data.has_more;
+    }
+
+    this.setState({
+      transactions: data.transactions,
+      loading: false,
+      firstHash,
+      lastHash,
+      hasAfter,
+      hasBefore,
+      queryParams,
+    });
+  }
+
+  /**
+   * Update transactions data state after requesting data from the server
+   */
+  getData = (queryParams) => {
+    hathorLib.walletApi.getSearchAddress(this.props.address, TX_COUNT, queryParams.hash, queryParams.page, (response) => {
+      if (response.success) {
+        this.handleDataFetched(response, queryParams);
+      }
+    });
+  }
+
+
+  render() {
+    if (this.state.transactions.length === 0) {
+      return <p>This address does not have any transactions yet.</p>
```

Comment:
Missing i18n markers.

Reply:
We don't have i18n in the explorer, should we do something?

Reply:
No. Sorry.